### PR TITLE
save support in source mode

### LINKED ISSUE #817: Save enabled only in design mode
## Are you reporting a feature request or a bug?

Bug

## [Check if the issue is already reported](http://docs.ckeditor.com/#!/guide/dev_issues_readme-section-avoid-duplicates)

https://dev.ckeditor.com/ticket/8354

## Provide detailed reproduction steps (if any)

1. Open editor and switch to source mode
2. Try saving using editor toolbar button

### Expected result

Saving with toolbar button should be possible in wysiwyg and source mode.

### Actual result
Saving with toolbar button is only possible in wysiwyg mode.

## Other details

* Browser: Any
* OS: Any
* CKEditor version: 3.0
* Installed CKEditor plugins: save

Initially this plugin was supposed to work in wysiwyg and source: 
https://dev.ckeditor.com/ticket/2836#comment:14
https://dev.ckeditor.com/attachment/ticket/2836/2836_6.patch
but it has most likely been made to work in wysiwyg only by mistake:
https://dev.ckeditor.com/ticket/8354
https://dev.ckeditor.com/changeset/3285

Except for this "won't fix" ticket https://dev.ckeditor.com/ticket/10902, I haven't found any problems related to it.

-----
Setting both modes to be dependent on form existence should fix this problem:
`command.modes = { wysiwyg: !!( editor.element.$.form ), source: !!( editor.element.$.form ) };`

### PR DESCRIPTION
**Pull request by wojtekw92**
## What is the purpose of this pull request?

bug fix

## Does your PR contain necessary tests?

All patches which change the editor code must include tests. You can always read more
on [PR testing](http://docs.ckeditor.com/#!/guide/dev_contributing_code-section-tests),
[how to set the testing environment](http://docs.ckeditor.com/#!/guide/dev_tests) and
[how to create tests](http://docs.ckeditor.com/#!/guide/dev_tests-section-creating-your-own-test)
in the official CKEditor documentation.

### This PR contains

- [X] Unit tests
- [x] Manual tests

## What changes did you make?

Added support for save button is source mode

closes #817

### CODE REVIEW

File: plugins/save/plugin.js [hunk 1]
```diff
@@ -45,7 +45,7 @@
 				return;
 
 			var command = editor.addCommand( pluginName, saveCmd );
-			command.modes = { wysiwyg: !!( editor.element.$.form ) };
+			command.modes = { wysiwyg: !!( editor.element.$.form ), source: !!( editor.element.$.form ) };
```
Comment (Comandeer) [inline plugins/save/plugin.js]: If at last we got to fix it, let's do it properly! I don't feel that `command.modes` is a proper place to do any of such checks (apparently [not only me](https://dev.ckeditor.com/ticket/8354#comment:1)). Modes should be just enabled and check should be moved e.g. to [`command.startDisabled`](https://docs.ckeditor.com/#!/api/CKEDITOR.commandDefinition-property-startDisabled).

File: tests/plugins/save/save.js [hunk 2]
```diff
@@ -1,9 +1,31 @@
 /* bender-tags: editor */
-/* bender-ckeditor-plugins: toolbar,save,wysiwygarea */
+/* bender-ckeditor-plugins: toolbar,save,wysiwygarea,sourcearea */
 
 bender.test( {
-	'test save event': function() {
-		var editor = CKEDITOR.replace( 'editor' ),
+	'test save event in WYSIWYG mode': function() {
```
Comment (Comandeer) [inline tests/plugins/save/save.js]: As code for tests is nearly identical, I'd think about creating some function to generate them or just to move common logic into it.

File: plugins/save/plugin.js [hunk 3]
```diff
@@ -44,8 +44,9 @@
 			if ( editor.elementMode != CKEDITOR.ELEMENT_MODE_REPLACE )
 				return;
 
+			saveCmd.startDisabled = !( editor.element.$.form );
```
Comment (Comandeer) [inline plugins/save/plugin.js]: This property should be added to `command`, not `saveCmd`. `saveCmd` is a template to create commands in all editors and `command` is an instance of that command in particular editor. We shouldn't modify the template, but the instance.

File: tests/plugins/save/save.js [hunk 4]
```diff
@@ -1,26 +1,35 @@
...
-		editor.on( 'instanceReady', function() {
-			editor.execCommand( 'save' );
+	editor.on( 'instanceReady', function() {
+		editor.execCommand( 'save' );
 
-			setTimeout( function() {
-				resume( function() {
-					assert.areSame( 1, count, 'save was fired once' );
-				} );
+		setTimeout( function() {
```
Comment (Comandeer) [inline tests/plugins/save/save.js]: I wonder if `afterCommandExec` could be used here instead of `setTimeout`.

File: plugins/save/plugin.js [hunk 5]
```diff
@@ -44,8 +44,9 @@
...
 			var command = editor.addCommand( pluginName, saveCmd );
-			command.modes = { wysiwyg: !!( editor.element.$.form ) };
+			command.modes = { wysiwyg: 1,source: 1 };
```
Comment (Comandeer) [inline plugins/save/plugin.js]: This property OTOH should be added to template (`saveCmd`) as it's shared across all commands created using that template.

File: tests/plugins/save/save.js (hunk 4 above)
Comment (wojtekw92) [reply tests/plugins/save/save.js]: unfortunately no, save plugin doesn't fire `afterCommandExec`

**Pull request merged**